manifests/config.pp

```diff
@@ -59,6 +59,38 @@
 
     }
 
+    'Suse': {
+      $pamconfig_mkhomedir_check_cmd  = "/usr/sbin/pam-config -q --mkhomedir | grep session:"
+      $pamconfig_check_cmd  = "/usr/sbin/pam-config -q --sss | grep session:"
+      $nsconfig_check_cmd = "/usr/bin/test $( /bin/grep -cE '(passwd|group|sudoers).*sss' /etc/nsswitch.conf ) -gt 2"
+      $nsconfig_cmd = "/bin/sed -i -e '/^\(passwd\|group\|sudoers\):/{/sss\b/ b;s/^\(.*\)$/\1 sss/}' /etc/nsswitch.conf"
+
+      if $mkhomedir {
+
+        exec { 'pam-config -a --mkhomedir':
+          path        => '/bin:/usr/bin:/sbin:/usr/sbin',
+          unless      => $pamconfig_mkhomedir_check_cmd,
+        }
+
+      }
+
+      exec { 'pam-config -a --sss':
+        path        => '/bin:/usr/bin:/sbin:/usr/sbin',
+        unless      => $pamconfig_check_cmd,
+      }
+
+      exec { 'echo "sudoers: file sss" >> /etc/nsswitch.conf':
```

Comment:
The problem is that all other distributions expect SLES are modifying the /etc/nsswitch.conf with the sssd package.
But i aggree with you. It would be better to use an existing nsswitch module.